bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -457,4 +491,82 @@ public static boolean isValidIPConfig(String ipAddress) {
         return false;
     }
 
+    private void scheduleToPollNetworkInterface(Map<String, Object> props) {
+
+        final NetUtil myself = this;
+        if (networkInterfacePollFuture != null) {
+            networkInterfacePollFuture.cancel(true);
+            networkInterfacePollFuture = null;
+        }
+
+        Runnable tempRunnable = new Runnable() {
+            @Override
+            public void run() {
+                myself.pollAndNotifyNetworkInterfaceAddress();
+            }
+        };
+
+        String pollFrequenceSecondsStr = "";
+        int pollFrequenceSeconds = INTERFACE_POLL_DEFAULT_FREQUENCY_SECONDS;
+        try {
+            pollFrequenceSecondsStr = (String) props.get(INTERFACE_POLL_FREQUENCY_SECONDS_KEY);
```

Comment:
This can cause a ClassCastException as properties might also be Numbers, not Strings.

In any case, you should move the code that reads the config parameters to the `modified` method, where all other params are processed as well.